packages/smooth_app/lib/pages/product/add_new_product/add_new_product_type.dart

```diff
@@ -0,0 +1,231 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/resources/app_icons.dart' as icons;
+import 'package:smooth_app/themes/smooth_theme_colors.dart';
+import 'package:smooth_app/themes/theme_provider.dart';
+
+class AddNewProductType extends StatefulWidget {
+  const AddNewProductType({
+    super.key,
+    required this.productType,
+    required this.checked,
+    required this.onChanged,
+  });
+
+  final ProductType productType;
+  final bool checked;
+  final void Function(ProductType) onChanged;
+
+  @override
+  State<AddNewProductType> createState() => _AddNewProductType();
+}
+
+class _AddNewProductType extends State<AddNewProductType>
+    with TickerProviderStateMixin {
+  AnimationController? _controller;
+  // ignore: use_late_for_private_fields_and_variables
+  Animation<Color?>? _colorAnimation;
+  // ignore: use_late_for_private_fields_and_variables
+  Animation<double>? _opacityAnimation;
```

Comment:
What about using "late" instead?

Reply:
This one is my fault.
Indeed we can use the `late` keyword, even if we reassign the value.